Unit-tests

itbel4onok opened the pull request:
- BunTest for test covering Bun class
- IngredientTest for test covering Ingredient class/ IngredientType
- BurgerMockTest, BurgerParamsTest, BurgerTest for test covering Burger class

itbel4onok committed:
Added classes:
- BunTest for test covering Bun class
- IngredientTest for test covering Ingredient class/ IngredientType
- BurgerMockTest, BurgerParamsTest, BurgerTest for test covering Burger class

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,50 @@
+    public void removeIngredientRemoveIngredientFromList() {
+        burger.addIngredient(ingredient);
+        burger.removeIngredient(0);
+        assertEquals(0, burger.ingredients.size());
```
Необходимо исправить: не следует использовать числа, значение которых не является очевидным (антипатерн магические числа). Лучше сохранить в переменную наименованием, отражающим суть содержимого.

nikolay-977 commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,44 @@
+import praktikum.IngredientType;
+
+@RunWith(Parameterized.class)
+public class IngredientTest {
```
Необходимо исправить: Напиши тесты для класса IngredientType в классе IngredientTypeTest. Например, можно использовать метод IngredientType.valueOf("SAUCE").

nikolay-977 commented on pom.xml:
```diff
@@ -13,4 +13,54 @@
         <maven.compiler.target>8</maven.compiler.target>
```
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из фалйа README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

itbel4onok committed:
Added classes:
- Add readme file
- Added IngredientType test
- Remove magical value from burger test
